src/Morphir/IR/Package.elm

```diff
@@ -231,6 +233,51 @@ eraseDefinitionAttributes def =
         |> mapDefinitionAttributes (\_ -> ()) (\_ -> ())
 
 
+{-| Filter package by List of List of QNames
+-}
+filterPackageByTypes : List QName -> Definition ta va -> Definition ta va
+filterPackageByTypes qNames pkgDef =
+    pkgDef.modules
+        |> Dict.toList
+        |> List.filterMap
+            (\( modName, modDef ) ->
+                let
+                    accControlledModDef =
+                        case modDef |> AccessControlled.withPublicAccess of
+                            Just mDef ->
+                                mDef
+
+                            Nothing ->
+                                Module.emptyDefinition
+
+                    moduleTypes : Dict Name (AccessControlled (Documented (Type.Definition ta)))
+                    moduleTypes =
+                        accControlledModDef.types
+                            |> Dict.toList
+                            |> List.filterMap
+                                (\( typName, typDef ) ->
+                                    if List.member (QName.fromTuple ( modName, typName )) qNames then
+                                        Just ( typName, typDef )
+
+                                    else
+                                        Nothing
+                                )
+                            |> Dict.fromList
+
+                    newModuleDefinitionTuple : ( Path, AccessControlled (Module.Definition ta va) )
+                    newModuleDefinitionTuple =
+                        ( modName, AccessControlled.public (Module.Definition moduleTypes Dict.empty) )
```

Comment:
This changes private modules to public and also wipes out values. Are you sure you want to do that?